### PR TITLE
Add generic handlers to Data Engine

### PR DESCRIPTION
**Pull request by alexandra-bucur**
Addressing https://github.com/odpi/egeria/issues/3843.

### CODE REVIEW

File: open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/builders/AssetBuilder.java [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.repositoryconnector.OMRSRepositoryHelper;
+import org.odpi.openmetadata.repositoryservices.ffdc.exception.TypeErrorException;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * AssetBuilder creates the parts of a root repository entity for an asset.
+ */
+public class AssetBuilder extends ReferenceableBuilder
```
Comment (popa-raluca) [inline open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/builders/AssetBuilder.java:18]: Unused class, can be removed

File: open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/DataEngineSchemaTypeHandler.java [hunk 2]
```diff
@@ -231,96 +239,130 @@ public void removeSchemaType(String userId, String schemaTypeGUID) throws Invali
...
+            InvalidParameterException {
         String methodName = "buildSchemaAttributeEntityDetail";
+        SchemaAttributeBuilder schemaAttributeBuilder = getSchemaAttributeBuilder(attribute);
 
-        SchemaAttributeBuilder builder = schemaTypeHandler.getSchemaAttributeBuilder(schemaAttribute);
+        return dataEngineCommonHandler.buildEntityDetail(schemaAttributeGUID, schemaAttributeBuilder.getInstanceProperties(methodName));
+    }
 
-        return dataEngineCommonHandler.buildEntityDetail(schemaAttributeGUID, builder.getInstanceProperties(methodName));
+    SchemaAttributeBuilder getSchemaAttributeBuilder(Attribute attribute) {
```
Comment (popa-raluca) [inline open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/DataEngineSchemaTypeHandler.java:309]: minCardinality. maxCardinality, defaultValueOverride, allowsDuplicateValues, orderedValues need to be set from Attribute bean

**Review by popa-raluca — APPROVED** [review]

File: open-metadata-implementation/access-services/data-engine/data-engine-api/src/main/java/org/odpi/openmetadata/accessservices/dataengine/model/Attribute.java [hunk 3]
```diff
@@ -27,6 +27,7 @@
     private String defaultValueOverride;
     private String dataType;
     private String defaultValue;
+    private String anchorGUID;
```
Comment (mandy-chessell) [inline open-metadata-implementation/access-services/data-engine/data-engine-api/src/main/java/org/odpi/openmetadata/accessservices/dataengine/model/Attribute.java:30]: Why is anchorGUID on the API?

File: open-metadata-implementation/access-services/data-engine/data-engine-api/src/main/java/org/odpi/openmetadata/accessservices/dataengine/model/Process.java [hunk 4]
```diff
@@ -18,7 +18,7 @@
 @JsonInclude(JsonInclude.Include.NON_NULL)
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
-public class Process extends Asset {
+public class Process {
```
Comment (mandy-chessell) [inline open-metadata-implementation/access-services/data-engine/data-engine-api/src/main/java/org/odpi/openmetadata/accessservices/dataengine/model/Process.java:21]: Will this break backward compatibility?

File: open-metadata-implementation/access-services/data-engine/data-engine-api/src/main/java/org/odpi/openmetadata/accessservices/dataengine/model/Process.java [hunk 5]
```diff
@@ -96,6 +104,102 @@ public void setParentProcesses(List<ParentProcess> parentProcesses) {
         this.parentProcesses = parentProcesses;
     }
 
+    public String getDisplayName() {
```
Comment (mandy-chessell) [inline open-metadata-implementation/access-services/data-engine/data-engine-api/src/main/java/org/odpi/openmetadata/accessservices/dataengine/model/Process.java:107]: Missing Javadoc

File: open-metadata-implementation/common-services/generic-handlers/src/main/java/org/odpi/openmetadata/commonservices/generichandlers/SchemaAttributeHandler.java [hunk 6]
```diff
@@ -717,4 +720,37 @@ public SCHEMA_ATTRIBUTE getSchemaAttribute(String userId,
...
+     * @throws UserNotAuthorizedException the user is not authorized to issue this request
+     * @throws PropertyServerException    there is a problem reported in the open metadata server(s)
+     */
+    public void updateSchemaAttribute(String userId,
+                                      String externalSourceGUID,
+                                      String externalSourceName,
+                                      String schemaAttributeGUID,
+                                      InstanceProperties instanceProperties) throws InvalidParameterException,
+                                                                                    PropertyServerException,
+                                                                                    UserNotAuthorizedException {
```
Comment (mandy-chessell) [inline open-metadata-implementation/common-services/generic-handlers/src/main/java/org/odpi/openmetadata/commonservices/generichandlers/SchemaAttributeHandler.java:741]: Code layout (method parameters and first brace)

Comment (alexandra-bucur) [reply open-metadata-implementation/common-services/generic-handlers/src/main/java/org/odpi/openmetadata/commonservices/generichandlers/SchemaAttributeHandler.java]: I missed that one :) now it should be fixed

File: open-metadata-implementation/access-services/data-engine/data-engine-api/src/main/java/org/odpi/openmetadata/accessservices/dataengine/model/Process.java (hunk 4 above)
Comment (alexandra-bucur) [reply open-metadata-implementation/access-services/data-engine/data-engine-api/src/main/java/org/odpi/openmetadata/accessservices/dataengine/model/Process.java:24]: As was discussed on the call, DE is still in development and we should not worry about backward compatibility for this development (as also other adjustments will be done)

File: open-metadata-implementation/access-services/data-engine/data-engine-api/src/main/java/org/odpi/openmetadata/accessservices/dataengine/model/Attribute.java (hunk 3 above)
Comment (alexandra-bucur) [reply open-metadata-implementation/access-services/data-engine/data-engine-api/src/main/java/org/odpi/openmetadata/accessservices/dataengine/model/Attribute.java:33]: Thank you for spotting the issue. Now I'm using the value.

**Review by popa-raluca — APPROVED** [review]

**Pull request merged**